### PR TITLE
[AUD-1413] Connect repeat and shuffle native buttons

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Description
* Moves queue to common store (minus sagas)

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

Ensured that store state showed correctly reflected repeat/shuffle values when pressing now playing buttons.

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rj-aud-1413-3

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rj-aud-1413-3

**Comment by sliptype** [comment]
This is merging into [rj-aud-1413-2](https://github.com/AudiusProject/audius-client/tree/rj-aud-1413-2), is that correct?

**Comment by raymondjacobson** [comment]
> This is merging into [rj-aud-1413-2](https://github.com/AudiusProject/audius-client/tree/rj-aud-1413-2), is that correct?

Wanted to land that and then I'll rebase

File: packages/mobile/src/components/animated-button/AnimatedButtonProvider.tsx [hunk 1]
```diff
@@ -186,9 +186,7 @@ const AnimatedButtonProvider = ({
     }
   }, [isDarkMode, setIconJSON, iconDarkJSON, iconLightJSON])
 
-  return iconJSON ? (
-    <AnimatedButton iconJSON={iconJSON} {...buttonProps} />
-  ) : null
+  return iconJSON && <AnimatedButton iconJSON={iconJSON} {...buttonProps} />
```
Comment (raymondjacobson) [inline packages/mobile/src/components/animated-button/AnimatedButtonProvider.tsx:189]: rebase should fix this!!!!

File: packages/web/src/store/queue/mobileSagas.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ import {
   getShuffleIndex,
   getShuffleOrder,
   getQueueAutoplay
-} from './selectors'
+} from '../../common/store/queue/selectors'
```
Comment (raymondjacobson) [inline packages/web/src/store/queue/mobileSagas.ts:34]: Fix import

File: packages/web/src/common/store/queue/selectors.ts [hunk 3]
```diff
@@ -9,30 +9,33 @@ import {
...
-export const getUndershot = (state: AppState) => state.queue.undershot
-export const getPositions = (state: AppState) => state.queue.positions
-export const getIndex = (state: AppState) => state.queue.index
-export const getRepeat = (state: AppState) => state.queue.repeat
-export const getQueueAutoplay = (state: AppState) => state.queue.queueAutoplay
-export const getShuffle = (state: AppState) => state.queue.shuffle
-export const getShuffleIndex = (state: AppState) => state.queue.shuffleIndex
-export const getShuffleOrder = (state: AppState) => state.queue.shuffleOrder
-export const getUidInQueue = (state: AppState, props: { uid: UID }) =>
+import { CommonState } from '..'
```
Comment (sliptype) [inline packages/web/src/common/store/queue/selectors.ts:12]:
```suggestion
import { CommonState } from 'common/store'
```

File: packages/mobile/src/components/now-playing-drawer/AudioControls.tsx [hunk 4]
```diff
@@ -69,6 +77,28 @@ export const AudioControls = () => {
...
+      dispatchWeb(shuffle({ enable: false }))
+    } else {
+      dispatchWeb(shuffle({ enable: true }))
+    }
+  }, [dispatchWeb, shuffleEnabled])
+
+  const onPressRepeat = useCallback(() => {
+    switch (repeatMode) {
+      case RepeatMode.ALL:
+        dispatchWeb(repeat({ mode: RepeatMode.SINGLE }))
```
Comment (sliptype) [inline packages/mobile/src/components/now-playing-drawer/AudioControls.tsx:91]: This is probably more expressive but I'm not a huge fan of calling dispatchWeb(repeat()) in each case (also in the shuffle handler above)

**Review by sliptype — APPROVED** [review]
lgtm!

File: packages/mobile/src/components/now-playing-drawer/AudioControls.tsx (hunk 4 above)
Comment (raymondjacobson) [reply packages/mobile/src/components/now-playing-drawer/AudioControls.tsx:91]: will change! reviewer is always right!

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rj-aud-1413-3

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rj-aud-1413-3

**Pull request merged**